crates/common/tedge_utils/src/fs_notify.rs

```diff
@@ -0,0 +1,441 @@
+use std::{
+    collections::{HashMap, HashSet},
+    hash::Hash,
+    path::{Path, PathBuf},
+};
+
+use async_stream::try_stream;
+pub use futures::{pin_mut, Stream, StreamExt};
+use inotify::{EventMask, Inotify, WatchMask};
+use strum_macros::Display;
+use try_traits::default::TryDefault;
+
+#[derive(Debug, Display, PartialEq, Eq, Clone, Hash)]
+pub enum Masks {
```

Comment:
The name `Masks` hardly convey the intent of this enum. Would `FileEvent` be better?